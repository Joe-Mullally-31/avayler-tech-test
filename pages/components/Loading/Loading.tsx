import { TailSpin } from 'react-loader-spinner';
import styles from "../../../styles/Loading.module.css";

export const Loading = () => (
  <div className={styles.loadingContainer}>
    <TailSpin
      height="80"
      width="80"
      color="#1e1e27"
      ariaLabel="Transaction data loading"
      radius="1"
      wrapperStyle={{
        justifyContent: 'center',
        paddingTop: '100px',
        columnFill: 'balance',
      }}
      wrapperClass=""
      visible={true}
    />
    Loading...
  </div>
);
