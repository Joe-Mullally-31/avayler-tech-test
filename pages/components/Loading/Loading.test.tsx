import { render } from '@testing-library/react';
import { Loading } from './Loading';

it('should render as expected', () => {
  expect(render(<Loading />)).toMatchSnapshot();
});
