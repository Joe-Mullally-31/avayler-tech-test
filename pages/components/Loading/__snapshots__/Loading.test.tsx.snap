// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="loadingContainer"
      >
        <div
          aria-busy="true"
          aria-label="Transaction data loading"
          class=""
          data-testid="tail-spin-loading"
          role="status"
          style="display: flex; justify-content: center; padding-top: 100px; column-fill: balance;"
        >
          <svg
            data-testid="tail-spin-svg"
            height="80"
            viewBox="0 0 38 38"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <lineargradient
                id="a"
                x1="8.042%"
                x2="65.682%"
                y1="0%"
                y2="23.865%"
              >
                <stop
                  offset="0%"
                  stop-color="#1e1e27"
                  stop-opacity="0"
                />
                <stop
                  offset="63.146%"
                  stop-color="#1e1e27"
                  stop-opacity=".631"
                />
                <stop
                  offset="100%"
                  stop-color="#1e1e27"
                />
              </lineargradient>
            </defs>
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <g
                transform="translate(1 1)"
              >
                <path
                  d="M36 18c0-9.94-8.06-18-18-18"
                  id="Oval-2"
                  stroke="#1e1e27"
                  stroke-width="2"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </path>
                <circle
                  cx="36"
                  cy="18"
                  fill="#fff"
                  r="1"
                >
                  <animatetransform
                    attributeName="transform"
                    dur="0.9s"
                    from="0 18 18"
                    repeatCount="indefinite"
                    to="360 18 18"
                    type="rotate"
                  />
                </circle>
              </g>
            </g>
          </svg>
        </div>
        Loading...
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="loadingContainer"
    >
      <div
        aria-busy="true"
        aria-label="Transaction data loading"
        class=""
        data-testid="tail-spin-loading"
        role="status"
        style="display: flex; justify-content: center; padding-top: 100px; column-fill: balance;"
      >
        <svg
          data-testid="tail-spin-svg"
          height="80"
          viewBox="0 0 38 38"
          width="80"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <lineargradient
              id="a"
              x1="8.042%"
              x2="65.682%"
              y1="0%"
              y2="23.865%"
            >
              <stop
                offset="0%"
                stop-color="#1e1e27"
                stop-opacity="0"
              />
              <stop
                offset="63.146%"
                stop-color="#1e1e27"
                stop-opacity=".631"
              />
              <stop
                offset="100%"
                stop-color="#1e1e27"
              />
            </lineargradient>
          </defs>
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <g
              transform="translate(1 1)"
            >
              <path
                d="M36 18c0-9.94-8.06-18-18-18"
                id="Oval-2"
                stroke="#1e1e27"
                stroke-width="2"
              >
                <animatetransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </path>
              <circle
                cx="36"
                cy="18"
                fill="#fff"
                r="1"
              >
                <animatetransform
                  attributeName="transform"
                  dur="0.9s"
                  from="0 18 18"
                  repeatCount="indefinite"
                  to="360 18 18"
                  type="rotate"
                />
              </circle>
            </g>
          </g>
        </svg>
      </div>
      Loading...
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
